eng/templates/Azure.ResourceManager.Template/tests/ProviderShortNameManagementTestBase.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.TestFramework;
+using NUnit.Framework;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.Template.Tests
+{
+    public class ProviderShortNameManagementTestBase : ManagementRecordedTestBase<ProviderShortNameManagementTestEnvironment>
+    {
+        protected ArmClient Client { get; private set; }
+
+        protected ProviderShortNameManagementTestBase(bool isAsync, RecordedTestMode mode)
+        : base(isAsync, mode)
+        {
+        }
+
+        protected ProviderShortNameManagementTestBase(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public void CreateCommonClient()
+        {
+            Client = GetArmClient();
+        }
+
+        protected async Task<ResourceGroupResource> CreateResourceGroup(SubscriptionResource subscription, string rgNamePrefix, AzureLocation location)
+        {
+            string rgName = Recording.GenerateAssetName(rgNamePrefix);
+            ResourceGroupData input = new ResourceGroupData(location);
+            var lro = await subscription.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, rgName, input);
+            return lro.Value;
+        }
+    }
+}
```

Comment:
what's the bellowing '-.-' file?

Reply:
Ah, I copied this file from the `SessionRecords` directory, just to avoid the `Scenario` directory being ingored because directories are implicitly in Git.

Reply:
Will the `-.-` file be checked in Git if we forget to delete it? If yes, we may add it to `.gitignore` file?